Spigot-API-Patches/0193-Expose-MinecraftServer-isRunning.patch

```diff
@@ -0,0 +1,47 @@
+From 0f38ed2b2376a7728b1251f5ef0fd80bde967e70 Mon Sep 17 00:00:00 2001
+From: JRoy <joshroy126@gmail.com>
+Date: Fri, 10 Apr 2020 21:24:35 -0400
+Subject: [PATCH] Expose MinecraftServer#isRunning
+
+This allows for plugins to detect if the server is actually turning off in onDisable rather than just plugins reloading.
+
+diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
+index b9973406..791ce81a 100644
+--- a/src/main/java/org/bukkit/Bukkit.java
++++ b/src/main/java/org/bukkit/Bukkit.java
+@@ -1661,6 +1661,15 @@ public final class Bukkit {
+     public static int getCurrentTick() {
+         return server.getCurrentTick();
+     }
++
++    /**
++     * Checks if the server is still running (not being shutdown).
++     *
++     * @return true if server is not in the process of being shutdown
++     */
++    public static boolean isServerRunning() {
++        return server.isServerRunning();
++    }
+     // Paper end
+ 
+     @NotNull
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index 80f9abdc..14c266fa 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -1455,5 +1455,12 @@ public interface Server extends PluginMessageRecipient {
+      * @return Current tick
+      */
+     int getCurrentTick();
++
++    /**
++     * Checks if the server is still running (not being shutdown).
++     *
++     * @return true if server is not in the process of being shutdown
++     */
++    boolean isServerRunning();
```

Comment:
Since this is on the Server interface. This seems redundant to me. `isRunning()` - I think would be sufficient.

Reply:
Fair

Reply:
Actually you can't have this because for some reason you're required to have the same method name in both classes

Reply:
Yes they must match. If you want it to be called isRunning, it must be the same on both.

Reply:
In that case, isServerRunning sounds better to make it clear what's running

Reply:
I'll let someone else weigh in from here. I would still rather it be isRunning than isServerRunning. But I am just one.

Reply:
I can appreciate wanting to be clear with `isServerRunning` however Bukkit/Server already sort of imply that, so I think I'm leaning towards `isRunning`. Open to thoughts.